packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -105,137 +128,135 @@ export const App = ({
-
-              <AccountRecoveryResetPassword path="/account_recovery_reset_password/*" />
-
-              <SigninReported path="/signin_reported/*" />
-              <SigninBounced
-                {...{ emailLookupComplete, email }}
-                path="/signin_bounced/*"
-              />
-              {/* Pages using the Ready view need to be accessible to logged out viewers,
-               * but need to be able to check if the user is logged in or logged out,
-               * so they are wrapped in this component.
-               */}
-              <PageWithLoggedInStatusState
-                Page={ResetPasswordConfirmed}
-                path="/reset_password_verified/*"
-              />
-              <PageWithLoggedInStatusState
-                Page={ResetPasswordWithRecoveryKeyVerified}
-                path="/reset_password_with_recovery_key_verified/*"
-              />
-              <PageWithLoggedInStatusState
-                Page={PrimaryEmailVerified}
-                path="/primary_email_verified/*"
-              />
-              <PageWithLoggedInStatusState
-                Page={SignupConfirmed}
-                path="/signup_verified/*"
-              />
-              <PageWithLoggedInStatusState
-                Page={SignupConfirmed}
-                path="/signup_confirmed/*"
-              />
-              <PageWithLoggedInStatusState
-                Page={SigninConfirmed}
-                path="/signin_verified/*"
-              />
-              <PageWithLoggedInStatusState
-                Page={SigninConfirmed}
-                path="/signin_confirmed/*"
-              />
-
-              <Confirm path="/confirm/*" {...{ sessionTokenId }} />
-              <ConfirmSignupCode path="/confirm_signup_code/*" />
-
-              <ThirdPartyAuthCallback path="/post_verify/third_party_auth/callback/*" />
-            </>
-          )}
-          <Settings path="/settings/*" {...{ showRecoveryKeyV2 }} />
-        </ScrollToTop>
-      </Router>
-    </>
+    <Router>
+      <WebChannelExample path="/web_channel_example/*" />
+
+      <CannotCreateAccount path="/cannot_create_account/*" />
+      <Clear path="/clear/*" />
+      <CookiesDisabled path="/cookies_disabled/*" />
+
+      <Legal path="/legal/*" />
+      <LegalTerms path="/legal/terms/*" />
+      <LegalTerms path="/:locale/legal/terms/*" />
+      <LegalPrivacy path="/legal/privacy/*" />
+      <LegalPrivacy path="/:locale/legal/privacy/*" />
+
+      <ResetPassword path="/reset_password/*" {...{ integration }} />
+      <ConfirmResetPassword
+        path="/confirm_reset_password/*"
+        {...{ integration }}
+      />
+      <CompleteResetPasswordContainer
+        path="/complete_reset_password/*"
+        {...{ integration }}
+      />
+
+      <LinkValidator
+        path="/account_recovery_confirm_key/*"
+        linkType={LinkType['reset-password']}
+        viewName="account-recovery-confirm-key"
+        getParamsFromModel={() => {
```

Comment:
Naming nit. Maybe just `getParams` would be better. I feel like more obviously maps to the `params` variable used below, and the params don't necessarily have to come from a model.